perceval/utils/statevector.py

```diff
@@ -401,19 +391,6 @@ def group_modes_simplification(self, group_size: int) -> BSDistribution:
         """
         simplified_distribution = BSDistribution()
         for bs, p in self.items():
-            bs_list = []
-            i, mode_sum = 0, 0
-            for mode in range(len(bs)):
-                if i == group_size: # max size of group is reached, create a new group
-                    bs_list.append(mode_sum)
-                    i, mode_sum = 0, 0
-                i += 1
-                mode_sum += bs[mode]
-            bs_list.append(mode_sum) # add the last group
-            bs = BasicState(bs_list)
-            if bs in simplified_distribution:
-                simplified_distribution[bs] = simplified_distribution[bs] + p
-            else:
-                simplified_distribution.add(bs, p)
-
+            bs = BasicState([sum([s for s in bs[group_size*k:group_size*(k+1)]]) for k in range(-(len(bs)//-group_size))]) # group modes by groups of size 'group_size'
```

Comment:
Can you add a comment to explain that the integer division here performs a ceil operation as this is not standard ?

Reply:
Also, have you checked what happens if the group size doesn't divide the number of modes ? I'm not sure the splicing here would work